### PR TITLE
feat: add `show_caption_above_media` parameter to `(Ext)Defaults`

### LINKED ISSUE #4306: Add `show_caption_above_media` parameter to `Defaults`
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

Currently when you want to set the `show_caption_above_media` parameter to every bot method that supports it (such as `send_video`, `send_photo`, `edit_message_caption`, etc.) you need to do so manually on each method call

### Describe the solution you'd like

It would be nice to set a default value for it via the `telegram.ext.Defaults` class, since you typically set it everywhere anyway just for consistency sake

### Describe alternatives you've considered

_No response_

### Additional context

I would also like to contribute for this feature myself if that's fine

### PR DESCRIPTION
**Pull request by servusdei2018**
Fixes #4306

## Summary

This patch introduces a default setting for the `show_caption_above_media` parameter in the `telegram.ext.Defaults` class. This enhancement streamlines the process of setting this parameter across various bot methods that support it, such as `send_video`, `send_photo`, and `edit_message_caption`.

## Changes Made

- A new property `show_caption_above_media` has been added to the `(Ext)Bot` class.

### CODE REVIEW

**Comment by servusdei2018** [comment]
cc @Poolitzer @Bibo-Joshi

**Comment by Bibo-Joshi** [comment]
Hi. Thanks for your PR. There is no need to ping us. We will reply when we have the time.

File: telegram/ext/_defaults.py [hunk 1]
```diff
@@ -114,6 +114,11 @@ async def main():
         do_quote(:obj:`bool`, optional): |reply_quote|
 
             .. versionadded:: 20.8
+
+        show_caption_above_media (:obj:`bool`, optional): Indicates whether the caption should be
+        shown above the media. Defaults to `None`.
```
Comment (Bibo-Joshi) [inline telegram/ext/_defaults.py:119]:
```suggestion
            shown above the media. Defaults to `None`.
```

File: telegram/ext/_extbot.py
Comment (Bibo-Joshi) [inline telegram/ext/_extbot.py:1]: this is not how we implement Defaults. Please have a look at the methods `_insert_defaults` and `_insert_defaults_for_ilq_results`. Note that changes will also be necessary for `Bot`. Finally, we will need new unit tests for this.

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Hi. Please see my comments below. Also, be sure that you have read our [contribution guide](https://docs.python-telegram-bot.org/en/v21.6/contributing.html) and include the check list in the PR description.

**Comment by krishna931074** [comment]
Stranger

**Comment by Bibo-Joshi** [comment]
closing due to inactivity. @servusdei2018 feel free to reopen anytime if you want to continue working on this

**Pull request closed without merging**